### DISCUSSION TITLE
netclient: Force reconnect upon network change

### DISCUSSION THREAD

**Discussion by gymnae** [gravitl/netmaker#1484] [Q&A]
**Problem statement**
I have a little Debian box behind my Internet Router, which has netclient installed as system service. When rebooting the box, it reconnects automatically to the netmaker network of which it is a peer.

But if the Debian box looses Wi-Fi and reconnects, or if the Internet router reboots, the service won't reconnect automatically, which requires manual intervention by logging in via SSH and restarting the service. This is an issue when I am away from home, since I neet the netmaker network to be able to SSH in. 

**Question**
Is it possible to configure netclient to re-connect, or restart itself as a service, whenever there's a network change on its host?

**Comment by mattkasun** [comment]
This should not be an issue.. I just tested by bring running a ping over wg. When I  disabling network interface, ping pauses as expected.   Re-establish network interface,  ping continues.

In the case where the public IP changes, the netclient daemon will detect the change and update configs accordingly.
